mne/encoding/model.py

```diff
@@ -0,0 +1,230 @@
+            else:
+                if self.mask_val is np.nan:
+                    mask = ~np.isnan(X)
+                else:
+                    mask = X != self.mask_val
+
+                # Decide which rows to remove
+                if self.mask_condition == 'all':
+                    # Remove rows w/ NO good values
+                    mask = mask.sum(axis=1) == 0
+                else:
+                    # Remove rows w/ AT LEAST 1 bad value
+                    mask = mask.sum(axis=1) < mask.shape[-1]
+
+            # Now change mask to rows that we wish to *keep*
+            self.mask = ~mask
+            mask_ixs = np.where(self.mask)[0]
+            mask_ixs = np.asarray(mask_ixs).squeeze()  # In case its a matrix
+            X = X[mask_ixs, :]
+            if y is not None:
+                y = y[mask_ixs, :]
+        else:
+            self.mask = np.ones(X.shape[0], dtype=bool)
+        # Return masked values
+        if y is not None:
+            return X, y
+        return X
+
+    def transform(self, X, y=None):
+        """Return a subset of data points.
+
+        Returns
+        -------
+        data : array, shape (n_features, n_ixs)
+            A subset of rows from the input array.
+        """
+        return self.mask_data(X, y)
+
+
+def get_final_est(est):
+    """Return the final component of a sklearn estimator/pipeline.
+
+    Parameters
+    ----------
+    est : pipeline, or sklearn / MNE estimator
+        An estimator chain from which you wish to pull the final estimator.
+
+    Returns
+    -------
+    est : the sklearn-style estimator at the end of the input chain.
+    """
+    # Define classes where we pull `est` manually
+    import sklearn
+    from distutils.version import LooseVersion
+    if LooseVersion(sklearn.__version__) < '0.16':
+        raise ValueError('Encoding models require sklearn version >= 0.16')
+
+    iter_classes = (SampleMasker,)
+    # Iterate in case we have meta estimators w/in meta estimators
+    while hasattr(est, '_final_estimator') or isinstance(est, iter_classes):
+        if isinstance(est, iter_classes):
+            est = est.est
+        else:
+            est = est._final_estimator
+    return est
+
+
+def get_coefs(est, coef_name='coef_'):
+    """Pull coefficients from an estimator object.
+
+    Parameters
+    ----------
+    est : a sklearn estimator
+        The estimator from which to pull the coefficients.
+    coef_name : string
+        The name of the attribute corresponding to the coefficients created
+        after fitting. Defaults to 'coef_'
+
+    Returns
+    -------
+    coefs : array, shape (n_targets, n_coefs)
+        The output coefficients.
+    """
+    if not hasattr(est, coef_name):
+        raise ValueError('Estimator either is not fit or does not use'
+                         ' coefficient name: %s' % coef_name)
+    coefs = getattr(est, coef_name)
+    return coefs
```

Comment:
This one could probably be private...it's mostly to make sure that a decent error gets thrown if the attribute doesn't exist. But we could just assume that the person knows what they're doing.